custom_components/tuya_local/devices/friedrich_uct08b10a_airconditioner

```diff
@@ -0,0 +1,520 @@
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: mode
+          conditions:
+            - dps_val: cold
+              value: cool
+            - dps_val: wet
+              value: dry
+            - dps_val: wind
+              value: fan_only
+            - dps_val: auto
+              value: auto
+            - dps_val: hot
+              value: heat
+              hidden: true
+    - id: 3
+      type: integer
+      name: current_temperature_in_c
+      optional: true
+      unit: C
+    - id: 23
+      name: current_temperature
+      type: integer
+      optional: true
+      hidden: true
+      range:
+        min: -20
+        max: 100
+    - id: 24
+      type: integer
+      name: temperature
+      optional: true
+      unit: F
+      range:
+        min: 690
+        max: 890
+      mapping:
+        - scale: 10
+          step: 10
+    - id: 2
+      name: temperature_in_c
+      type: integer
+      optional: true
+      range:
+        min: 205
+        max: 317
+      mapping:
+        - scale: 10
+          step: 5
+      unit: C
+    - id: 4
+      type: string
+      name: mode
+      hidden: true
+    - id: 5
+      name: fan_mode
+      type: string
+      mapping:
+        # the unit only truly supports 4 speeds: low, med, high, auto
+        - dps_val: auto
+          value: auto
+        - dps_val: low
+          value: low
+        - dps_val: mid
+          value: medium
+        - dps_val: high
+          value: high
+        # the below are all duplicates of some of the above and don't change air speed
+        - dps_val: mute
+          value: quiet
+          optional: true
+          hidden: true
+        - dps_val: mid_low
+          value: medlow
+          optional: true
+          hidden: true
+        - dps_val: mid_high
+          value: medhigh
+          optional: true
+          hidden: true
+        - dps_val: strong
+          value: strong
+          optional: true
+          hidden: true
+    - id: 123
+      name: preset_mode
+      # Tuya IOT dev says this is "bool" but the values aren't boolean
+      type: string
+      mapping:
+        - dps_val: "0018"
+          value: comfort
+        - dps_val: "0019"
+          value: eco
+        - dps_val: null
+          value: comfort
+          hidden: true
+    - id: 128
+      name: style
+      type: string
+      mapping:
+        - dps_val: "0"
+          value: wall
+        - dps_val: "1"
+          value: floor
+        - dps_val: "2"
+          value: missing mapping
+        - dps_val: "3"
+          value: through-the-wall
+    - id: 8
+      name: legacy_preset_mode
+      type: boolean
+      # the unit's actual hardware doesn't support this functionality even though the firmware / DP is exposed
+      # we are forced to add it so that Tuya Local gets to 100% DP match and allows us to pick this YAML for this unit
+      hidden: true
+      optional: true
+      mapping:
```

Comment:
I wouldn't bother with the mapping if it is just here for matching purposes. But as it is set to be optional, it is not even useful for that.